Lint port table ('AvailablePorts.md')

BenWiederhake opened the pull request:
As requested by popular demand ;) https://github.com/SerenityOS/serenity/pull/5325#discussion_r575657614

nico commented on Meta/lint-ports.py:
```diff
@@ -0,0 +1,67 @@
+import re
+
+
+PORT_TABLE_REGEX = re.compile('^\\| \\[`([^`]+)`\\][^`]+$')
```
nits:

1. Use a raw string (`r'^\| \[`([^`]+)`\][^`]+$'`)
2. Add a comment like `# Matches e.g. "| [``bash``]..." and captures "bash" in group 1

nico commented on Meta/lint-ports.py:
```diff
@@ -0,0 +1,67 @@
+            match = PORT_TABLE_REGEX.match(line)
+            if not match:
+                continue
+            ports.add(match.groups()[0])
```
nit: `match.group(1)`

But since it's a single group, I think the whole loop body can likely be replaced with `ports.add(PORT_TABLE_REGEX.findall(line))`

nico replied:
If you use `re.M` in options, I think you don't even need to go line-by-line. Just

```
  return set(PORT_TABLE_REGEX.findall(open(filename, 'r').read()))
```

might be enough as function body.

BenWiederhake replied:
I'll use `with open()`, to avoid leaking the fd. It doesn't really matter here, but [I like cleaning up fds](#5342). :)

BenWiederhake commented:
Changes since last push:
- Incorporated all the suggestions – thanks! I didn't know that `re` has such a convenient interface :D
- Rebased for convenience
